packages/rest/src/__tests__/integration/rest.server.integration.ts

```diff
@@ -1220,6 +1220,57 @@ paths:
     await server.stop();
   });
 
+  it('registers handler routes after start and getApiSpec', async () => {
+    const server = await givenAServer();
+    await server.start();
+    const client = createClientForHandler(server.requestHandler);
+    // No `/greet` is present
+    await client.get('/greet').expect(404);
+
+    // Add DummyController after server.start
+    const greetSpec = {
+      responses: {
+        200: {
+          content: {'text/plain': {schema: {type: 'string'}}},
+          description: 'greeting of the day',
+        },
+      },
+    };
+
+    server.route('get', '/greet', greetSpec, function greet() {
+      return 'Hello';
+    });
+
+    // Now `/greet` is available
+    await client.get('/greet').expect(200, 'Hello');
+
+    // Now update api specs
+    await server.getApiSpec();
```

Comment:
a bit confused here, why have to call `await server.getApiSpec()` before binding, instead of after?

same question for the next test when unbind the route.

Reply:
It's to verify that getApiSpec() won't mess up the rebuild of http handler. The original issue was caused by that.